src/Access/RowPolicyCache.cpp

```diff
@@ -215,40 +218,97 @@ void RowPolicyCache::mixFiltersFor(EnabledRowPolicies & enabled)
         std::vector<RowPolicyPtr> policies;
     };
 
-    std::unordered_map<MixedFiltersKey, MixerWithNames, Hash> mixers;
+    std::unordered_map<MixedFiltersKey, MixerWithNames, Hash> table_mixers;
+    std::unordered_map<MixedFiltersKey, MixerWithNames, Hash> database_mixers;
+
 
+    // prepare database_mixers
```

Comment:
This piece of preparing mixers begs for more detailed comment. Please add some explanation of what is going on and why, while you still understand it.

Reply:
The comment "prepare database_mixers" seems comprehensive in conjunction with the comment "Accumulates filters from multiple row policies and joins them using the AND logical operation" for class FiltersMixer.
(which is outdated by the way: not only 'AND', but also 'OR').

Anyway, I'll try to invent something.

In the loop we effectively fill mixers for all rules specified per an entire database.
In the next loop we fill mixers for specific tables, using database mixers.

This code looks like a duplication, but all attempts to improve the situation reduced readability so I gave up.